clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -783,7 +784,13 @@ public void unsubscribe(TopicPartition... partitions) {
     private void scheduleAutoCommitTask(final long interval) {
         DelayedTask task = new DelayedTask() {
             public void run(long now) {
-                commit(CommitType.ASYNC);
+                commitAsync(new OffsetCommitCallback() {
```

Comment:
I think some application actually will do both - nothing prevents them from doing that. I am slightly preferring having auto commit logged separately for user experience and facilitate debugging, given it is only tiny amount of code.